spec/models/dialog_spec.rb

```diff
@@ -82,18 +82,32 @@
   end
 
   context "#destroy" do
+    let(:dialog) { FactoryBot.create(:dialog) }
+
     it "destroy without resource_action association" do
-      expect(FactoryBot.create(:dialog).destroy).to be_truthy
+      expect(dialog.destroy).to be_truthy
       expect(Dialog.count).to eq(0)
     end
 
-    it "destroy with resource_action association" do
-      dialog = FactoryBot.create(:dialog)
+    it "does not destroy dialog with resource_action association" do
       FactoryBot.create(:resource_action, :action => "Provision", :dialog => dialog, :resource_type => Dialog, :resource_id => dialog.id)
-      expect { dialog.destroy }
-      .to raise_error(RuntimeError, "Dialog cannot be deleted because it is connected to other components: [\"Dialog:#{dialog.id} - #{dialog.name}\"]")
+      dialog.destroy
       expect(Dialog.count).to eq(1)
     end
+
+    describe "#reject_if_has_resource_actions" do
+      it "invokes from before_destroy callback" do
```

Comment:
We don't need to test that rails callbacks are invoked.  Instead, would you mind adding a test for the `:prepend => true` to validate that dependent records aren't destroyed if the callback throws `:abort`?

Reply:
yes, it is extra

Reply:
it was added to clarify next expectation: it is not clear in the next expectation that `dialog.send(:reject_if_has_resource_actions)` would be invoked from callback and text of error message out of `before_destroy` would still the same as before changes.

Reply:
test modified, and discussed :prepend => true` removed